target/linux/mediatek/dts/mt7622-xiaomi-redmi-router-ax6s.dts

```diff
@@ -0,0 +1,342 @@
+		mux {
+			function = "pwm";
+			groups = "pwm_ch7_2";
+		};
+	};
+
+	/* Serial NAND is shared pin with SPI-NOR */
+	serial_nand_pins: serial-nand-pins {
+		mux {
+			function = "flash";
+			groups = "snfi";
+		};
+	};
+
+	uart0_pins: uart0-pins {
+		mux {
+			function = "uart";
+			groups = "uart0_0_tx_rx" ;
+		};
+	};
+
+	watchdog_pins: watchdog-pins {
+		mux {
+			function = "watchdog";
+			groups = "watchdog";
+		};
+	};
+};
+
+&eth {
+	pinctrl-names = "default";
+	pinctrl-0 = <&eth_pins>;
+	status = "okay";
+
+	gmac0: mac@0 {
+		compatible = "mediatek,eth-mac";
+		reg = <0>;
+
+		phy-connection-type = "2500base-x";
+
+		fixed-link {
+			speed = <2500>;
+			full-duplex;
+			pause;
+		};
+	};
+
+	mdio-bus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		switch@0 {
+			compatible = "mediatek,mt7531";
+			reg = <0>;
+			reset-gpios = <&pio 54 GPIO_ACTIVE_HIGH>;
+
+			ports {
+				#address-cells = <1>;
+				#size-cells = <0>;
+
+				wan: port@1 {
+					reg = <1>;
+					label = "wan";
+				};
+
+				port@2 {
+					reg = <2>;
+					label = "lan1";
+				};
+
+				port@3 {
+					reg = <3>;
+					label = "lan2";
+				};
+
+				port@4 {
+					reg = <4>;
+					label = "lan3";
+				};
+
+				port@6 {
+					reg = <6>;
+					label = "cpu";
+					ethernet = <&gmac0>;
+					phy-mode = "2500base-x";
+
+					fixed-link {
+						speed = <2500>;
+						full-duplex;
+						pause;
+					};
+				};
+			};
+		};
+	};
+};
+
+&snand {
+	pinctrl-names = "default";
+	pinctrl-0 = <&serial_nand_pins>;
+	status = "okay";
+
+	mediatek,bmt-v2;
+	mediatek,bmt-table-size = <0x1000>;
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partition@0 {
+			label = "Preloader";
+			reg = <0x0 0x80000>;
+			read-only;
+		};
+
+		partition@80000 {
+			label = "ATF";
+			reg = <0x80000 0x40000>;
+			read-only;
+		};
+
+		partition@c0000 {
+			label = "u-boot";
+			reg = <0xc0000 0x80000>;
+			read-only;
+		};
+
+		partition@140000 {
+			label = "u-boot-env";
+			reg = <0x140000 0x40000>;
+		};
+
+		partition@180000 {
+			label = "bdata";
+			reg = <0x180000 0x40000>;
+			read-only;
+		};
+
+		factory: partition@1c0000 {
+			label = "factory";
+			reg = <0x1c0000 0x80000>;
+			read-only;
+		};
+
+		partition@240000 {
+			label = "crash";
+			reg = <0x240000 0x40000>;
+		};
+
+		partition@280000 {
+			label = "crash_log";
+			reg = <0x280000 0x40000>;
+		};
+
+		partition@2c0000 {
+			label = "firmware";
+			compatible = "denx,fit";
+			openwrt,cmdline-match = "firmware=0";
+			reg = <0x2c0000 0x1e00000>;
+		};
+
+		partition@20c0000 {
+			label = "firmware1";
+			compatible = "denx,fit";
+			openwrt,cmdline-match = "firmware=1";
+			reg = <0x20c0000 0x1e00000>;
+		};
+
+		/* ubi partition is the result of squashing
+		 * next consecutive stock partitions:
+		 * - overlay (ubi overlay)
+		 * - obr
+		 * resulting ~56MiB space for packages.
+		 */
+
+		partition@3ec0000 {
+			label = "ubi";
+			reg = <0x3ec0000 0x3700000>;
+		};
```

Comment:
To completely disable `Dual Boot`, just run the following commands:
```
nvram set "boot_fw1=run boot_rd_img;bootm"
nvram commit
```

Reply:
I don't remember if tftp recovery would restore those variables, so it could potentially complicate or degrade the experience for those who reverted to stock later...

Of course, one could always just tell people to reset boot_fw1 to `boot_fw1=run boot_rd_img2;bootm` in the revert to stock instructions too.

It's a bit of a trade-off really. A couple less install/revert instructions + some vague dual-firmware resilience vs. ~30mib extra package space.

We were happy to sacrifice dual boot on ramips for the extra space (If I recall, it was quite a bit less free space regained there as well), but what about here? 

@dangowrt Opinion? I'm quite happy to go either way in regards to storage use on the stock-alike layout.

Reply:
> I don't remember if tftp recovery would restore those variables, so it could potentially complicate or degrade the experience for those who reverted to stock later...

I have already tested this case. The bootloader after installing the firmware restores the default values (including `boot_fw1`). 
By the way, after that `ethaddr` was changed to the one that is hardcoded in bootloader (`ethaddr=00:0C:E7:11:22:33 `).

Reply:
Tested the tftp recovery yesterday, it's the same ip-as-hex scheme described on the AX3600 wiki page. However, it did not reset my ethaddr in the manner you described. I'll probably end up checking for it in the network script and then just offset it from the wan/label address if it happens, so things don't break with multiple units on the same network.

Or even easier yet, just decrement by 1 from the 2.4ghz radio mac address, as that results in the same mac address. (At least on my unit anyway)